Crash after installation

#### General Information

* **OpenLauncher Version:** 220103
* **Android Version:** 5.1.1
* **Custom ROM:** None
* **Install Source:** built from source (master branch)
* **Device:** Google AVD

#### Description
App crashes immediately after start. It works fine on this commit https://github.com/OpenLauncherTeam/openlauncher/commit/513acc13cf6b78584bd7caeb5faa43ede74e40b0 Probably caused by 0ac9f7065d741174d1c967f2c89ba62b2d589a81

#### Log

```
Build version: 220103 
Build date: 1981-01-01 01:01:02 
Current date: 2022-01-03 22:46:56 
Device: Unknown Android SDK built for x86 
 
Stack trace:  
java.lang.RuntimeException: An error occured while executing doInBackground()
	at android.os.AsyncTask$3.done(AsyncTask.java:304)
	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:355)
	at java.util.concurrent.FutureTask.setException(FutureTask.java:222)
	at java.util.concurrent.FutureTask.run(FutureTask.java:242)
	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
	at java.lang.Thread.run(Thread.java:818)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String android.content.ComponentName.getPackageName()' on a null object reference
	at com.benny.openlauncher.util.AppManager$AsyncGetApps.doInBackground(AppManager.java:185)
	at android.os.AsyncTask$2.call(AsyncTask.java:292)
	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
	... 4 more
```

#### Steps to reproduce bug

1. Install OpenLauncher 
2. Press HOME button

--- a/app/src/main/java/com/benny/openlauncher/util/AppManager.java
+++ b/app/src/main/java/com/benny/openlauncher/util/AppManager.java
@@ -9,7 +9,9 @@ import android.content.pm.ResolveInfo;
 import android.content.pm.ShortcutInfo;
 import android.os.AsyncTask;
 import android.os.Build;
+import android.os.Process;
 import android.os.UserHandle;
+import android.os.UserManager;
 import android.support.annotation.NonNull;
 
 import com.benny.openlauncher.activity.HomeActivity;
@@ -177,13 +179,27 @@ public class AppManager {
                         nonFilteredAppsTemp.add(app);
                     }
                 }
+            } else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N_MR1) {
+                UserManager userManager = (UserManager) _context.getSystemService(Context.USER_SERVICE);
+                // LauncherApps.getProfiles() is not available for API 25, so just get all associated user profile handlers
+                List<UserHandle> profiles = userManager.getUserProfiles();
+                LauncherApps launcherApps = (LauncherApps) _context.getSystemService(Context.LAUNCHER_APPS_SERVICE);
+                for (UserHandle userHandle : profiles) {
+                    List<LauncherActivityInfo> apps = launcherApps.getActivityList(null, userHandle);
+                    for (LauncherActivityInfo info : apps) {
+                        List<ShortcutInfo> shortcutInfo = Tool.getShortcutInfo(getContext(), info.getComponentName().getPackageName());
+                        App app = new App(_packageManager, info, shortcutInfo);
+                        app._userHandle = userHandle;
+                        LOG.debug("adding work profile to non filtered list: {}, {}, {}", app._label, app._packageName, app._className);
+                        nonFilteredAppsTemp.add(app);
+                    }
+                }
             } else {
                 Intent intent = new Intent(Intent.ACTION_MAIN, null);
                 intent.addCategory(Intent.CATEGORY_LAUNCHER);
                 List<ResolveInfo> activitiesInfo = _packageManager.queryIntentActivities(intent, 0);
                 for (ResolveInfo info : activitiesInfo) {
-                    List<ShortcutInfo> shortcutInfo = Tool.getShortcutInfo(getContext(), intent.getComponent().getPackageName());
-                    App app = new App(_packageManager, info, shortcutInfo);
+                    App app = new App(_packageManager, info, null);
                     LOG.debug("adding app to non filtered list: {}, {}, {}", app._label,  app._packageName, app._className);
                     nonFilteredAppsTemp.add(app);
                 }

